hs_composite_resource/models.py

```diff
@@ -127,6 +212,145 @@ def can_set_folder_to_fileset(self, dir_path):
         # if there are any files in the dir_path, we can set the folder to fileset aggregation
         return len(files_in_path) > 0
 
+    def can_set_folder_to_model_instance_aggregation(self, dir_path):
```

Comment:
This type of logic does not belong in composite resource, all the way down to line 353.  The aggregations should be able to decide if it can set the folder to an aggregation.

Reply:
Moved this code logic to logical file classes.